util/pull/pullprogress/progress.go

```diff
@@ -106,6 +106,7 @@ func trackProgress(ctx context.Context, desc ocispecs.Descriptor, manager PullMa
 		select {
 		case <-ctx.Done():
 			onFinalStatus = true
+			ctx = context.TODO()
```

Comment:
Could you add a comment line to explain this?

Reply:
This was handled in https://github.com/moby/buildkit/pull/3443